developer-hub/src/reference_docs/extrinsic_encoding.rs

```diff
@@ -0,0 +1,172 @@
+//!
+//! ```text
+//! version_and_maybe_signature = concat(
+//!     version_and_signed,
+//!     from_address,
+//!     signature,
+//!     signed_extensions_extra,
+//! )
+//! ```
+//!
+//! Each of the details to be concatenated together is explained below:
+//!
+//! ### version_and_signed
+//!
+//! This is one byte, equal to `0x84` or `0b1000_0100` (ie an upper 1 bit to denote that it is
+//! signed, and then the transaction version, 4, in the lower bits).
+//!
+//! ### from_address
+//!
+//! This is the [SCALE encoded][frame::deps::codec] address of the sender of the extrinsic. The
+//! address is the first generic parameter of [`sp_runtime::generic::UncheckedExtrinsic`], and so
+//! can vary from chain to chain.
+//!
+//! The address type used on the Polkadot relay chain is [`sp_runtime::MultiAddress<AccountId32>`],
+//! where `AccountId32` is defined [here][`sp_core::crypto::AccountId32`]. When constructing a
+//! signed extrinsic to be submitted to a Polkadot node, you'll always use the
+//! [`sp_runtime::MultiAddress::Id`] variant.
+//!
+//! ### signature
+//!
+//! This is the [SCALE encoded][frame::deps::codec] signature. The signature type is configured via
+//! the third generic parameter of [`sp_runtime::generic::UncheckedExtrinsic`], which determines the
+//! shape of the signature and signing algorithm that should be used.
+//!
+//! The signature is obtained by signing the _signed payload_ bytes (see below on how this is
+//! constructed) using the private key associated with the address and correct algorithm.
+//!
+//! The signature type used on the Polkadot relay chain is [`sp_runtime::MultiSignature`]; the
+//! variants there are the types of signature that can be provided.
+//!
+//! ### signed_extensions_extra
+//!
+//! This is the concatenation of the [SCALE encoded][frame::deps::codec] bytes representing each of
+//! the [_signed extensions_][sp_runtime::traits::SignedExtension], and are configured by the
+//! fourth generic parameter of [`sp_runtime::generic::UncheckedExtrinsic`]. Signed extensions are,
+//! briefly, a means for different chains to extend the "basic" extrinsic format with custom data
+//! that can be checked by the runtime.
+//!
+//! When it comes to constructing an extrinsic, each signed extension has two things that we are
+//! interested in here:
+//!
+//! - The actual SCALE encoding of the signed extension type itself; this is what will form our
+//!   `signed_extensions_extra` bytes.
+//! - An `AdditionalSigned` type. This is SCALE encoded into the `signed_extensions_additional`
+//!   data of the _signed payload_ (see below).
+//!
+//! Either (or both) of these can encode to zero bytes.
+//!
+//! Each chain configures the set of signed extensions that it uses in its runtime configuration.
+//! At the time of writing, Polkadot configures them
+//! [here](https://github.com/polkadot-fellows/runtimes/blob/1dc04eb954eadf8aadb5d83990b89662dbb5a074/relay/polkadot/src/lib.rs#L1432C25-L1432C25).
```

Comment:
I think it is good that you are using Polkadot as reference, but it would be good to mention that generally you can obtain all of this info either by reading the code in a static way, or from the metadata in a dynamic way. 

btw, do you want to also make a placeholder for a metadata ref-doc and do it? :D

Reply:
I wonder whether metadata needs a ref-doc, or whether it should just be properly documented in the `frame-metadata` crate? Maybe both; might be worth starting with adding good docs to `frame-metadata` and then writing a ref doc if there turns out to be some more general documentation about metadata that would conceptually span several crates?

But yeah, I'm happy to write about this too!